### PR TITLE
refactor: rewrite orga_data_entry into Vue components

### PR DESCRIPTION
**Pull request by sakutademos**
### Ticket
[DPLAN-2825](https://demoseurope.youtrack.cloud/issue/DPLAN-2825/Hausnummer-ist-weit-ins-Abseits-geraten)

**Description:** This PR creates Vue components to replace the inline Twig templates. The challenge here is that this template is used by every demosplan project and the projects may have a different layout. For example some projects use an accordion to display an organisation data, others show only brief information, while others show the detailed information. 

There will be additional PRs for adjustments specific to each project.

Better not to include it in the release, as it needs to be tested first... Merge into the main branch after CF

### CODE REVIEW

File: client/js/components/user/orgaDataEntry/OrganisationDataForm.vue [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+  computed: {
+    displaySlug () {
+      return hasPermission('feature_orga_slug') &&
+        !hasPermission('feature_orga_slug_edit') &&
+        this.organisation.currentSlugName !== ''
+    },
+
+    displayCustomer () {
+      return hasPermission('feature_display_customer_names') &&
+        this.customers && this.customers.length > 0
```
Comment (salisdemos) [inline client/js/components/user/orgaDataEntry/OrganisationDataForm.vue:351]:
```suggestion
      return hasPermission('feature_display_customer_names') && this.customers?.length > 0
```

File: client/js/components/user/orgaDataEntry/OrganisationDataForm.vue [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+          class="mb-2"
+          v-model="organisation.phone"
+          data-cy="organisationData:phone"
+          :name="`${organisation.id}:address_phone`"
+          :label="{
+            text: Translator.trans('phone')
+          }"
+          :disabled="!isOrgaDataEditable" />
+
+        <!-- TODO: show all types instead of disabled select-->
```
Comment (salisdemos) [inline client/js/components/user/orgaDataEntry/OrganisationDataForm.vue:212]: still a todo?

**Review by salisdemos — CHANGES_REQUESTED** [review]
Thanks. Great PR. And thanks for using TW classes.
I just have some minor comments.

But I have to mention:
I can't tell if the logic is still correct and I dind't checked the Interface. Codereview only. 

Am I right? Can we remove the Project-Overwrites whit this PR?
Is it correct, that You want to merge that into release?

**Comment by sakutademos** [comment]
> Am I right? Can we remove the Project-Overwrites whit this PR? Is it correct, that You want to merge that into release?

1. For now I cannot remove the project-overrides because many projects have differences: `Accordion` for displaying organisation data, or displaying detailed or short organisation data. Do we want to standardize this for all projects? If so, I can refactor and remove the overrides, but there are still projects where the `public_display`, `notification_settings`, or `paper_copy` blocks are not shown. Each project has its own adjustments... I am not sure what is better approach here.. you can check the last file here `orga_data_entry` in templates, there you can see the idea.. it is not the one component, but it is divided into components and blocks..

2. no, I do not want to include it in the release because we have a CF today and 4 projects need to be delivered to customers. It's too risky with too little time. It will be merged into the main branch after the CF

**Comment by salisdemos** [comment]
> 2. no, I do not want to include it in the release because we have a CF today and 4 projects need to be delivered to customers. It's too risky with too little time. It will be merged into the main branch after the CF

I just wondered, because You change the target branch to release.

**Comment by sakutademos** [comment]
> > 2. no, I do not want to include it in the release because we have a CF today and 4 projects need to be delivered to customers. It's too risky with too little time. It will be merged into the main branch after the CF
> 
> I just wondered, because You change the target branch to release.

I mistakenly merged a release branch instead of the main branch.. thats why i changed the goal branch here.. sorry for the confusion

File: client/js/components/user/orgaDataEntry/PaperCopyPreferences.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  },
+
+  computed: {
+    paperCopyCountOptions () {
+      return Array.from({ length: 11 }, (_, i) => ({
+        label: i.toString(),
+        value: i,
+      }))
+    },
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/PaperCopyPreferences.vue]: right, I placed it into methods https://github.com/demos-europe/demosplan-core/pull/3810/commits/905c2a9da224fc1ee3bb1de3b2d3f6e9fb3e782e

I am not sure which option is better, but the first option for me is more readable :D

File: client/js/components/user/orgaDataEntry/PaperCopyPreferences.vue [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        :hint="Translator.trans('explanation.organisation.copies.kind')" />
+    </div>
+
+    <div
+      v-if="hasPermission('field_organisation_competence')"
+      class="w-full mb-3">
+      <dp-text-area
+        id="orga_competence"
+        data-cy="organisationData:paperCopy:competence"
+        :name="`${organisation.id || ''}:competence`"
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/PaperCopyPreferences.vue]: thx for spotting this (it cannot be undefined), done here https://github.com/demos-europe/demosplan-core/pull/3810/commits/905c2a9da224fc1ee3bb1de3b2d3f6e9fb3e782e

File: client/js/components/user/orgaDataEntry/OrganisationDataForm.vue (hunk 1 above)
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/OrganisationDataForm.vue]: thx, done here https://github.com/demos-europe/demosplan-core/pull/3810/commits/905c2a9da224fc1ee3bb1de3b2d3f6e9fb3e782e

File: client/js/components/user/orgaDataEntry/OrganisationDataForm.vue [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+  components: {
+    DpRadio,
+    DpInput,
+    DpTextArea,
+    DpSelect,
+    DpEditor,
+    DpCheckbox
+  },
+
+  props: {
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/OrganisationDataForm.vue:279]: done here https://github.com/demos-europe/demosplan-core/pull/3810/commits/905c2a9da224fc1ee3bb1de3b2d3f6e9fb3e782e

File: client/js/components/user/orgaDataEntry/OrganisationDataForm.vue [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+export default {
+  name: 'OrganisationDataForm',
+
+  components: {
+    DpRadio,
+    DpInput,
+    DpTextArea,
+    DpSelect,
+    DpEditor,
+    DpCheckbox
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/OrganisationDataForm.vue]: thx, done here https://github.com/demos-europe/demosplan-core/pull/3810/commits/905c2a9da224fc1ee3bb1de3b2d3f6e9fb3e782e

File: client/js/bundles/user/editOrga.js [hunk 7]
```diff
@@ -13,11 +13,20 @@
 
 import { DpAccordion, DpEditor, dpValidate } from '@demos-europe/demosplan-ui'
 import { initialize } from '@DpJs/InitVue'
+import OrganisationDataForm from '@DpJs/components/user/orgaDataEntry/OrganisationDataForm'
+import EmailNotificationSettings from '@DpJs/components/user/orgaDataEntry/EmailNotificationSettings'
+import PaperCopyPreferences from '@DpJs/components/user/orgaDataEntry/PaperCopyPreferences'
+import OrganisationBrandingSettings from '@DpJs/components/user/orgaDataEntry/OrganisationBrandingSettings'
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/bundles/user/editOrga.js]: thx, done here https://github.com/demos-europe/demosplan-core/pull/3810/commits/905c2a9da224fc1ee3bb1de3b2d3f6e9fb3e782e

File: client/js/components/user/orgaDataEntry/PaperCopyPreferences.vue [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    id="paperCopy"
+    class="w-3/4">
+    <legend class="font-size-large weight--normal u-mb-0_75">
+      {{ Translator.trans('copies.paper') }}
+    </legend>
+
+    <div
+      v-if="hasPermission('field_organisation_paper_copy')"
+      class="w-full mb-3">
+      <!-- TODO: create PR in demosplan-ui -->
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/PaperCopyPreferences.vue:14]: done

File: client/js/components/user/orgaDataEntry/OrganisationDataForm.vue (hunk 2 above)
Comment (sakutademos) [reply client/js/components/user/orgaDataEntry/OrganisationDataForm.vue]: I have to discuss the display of the types with the PO, but for now its done

**Review by salisdemos — APPROVED** [review]
Looks really good! Thank You!

Please just change the target branch :pray:

**Pull request merged**